test/serde_arrow_primitive_array_SUITE.erl

```diff
@@ -0,0 +1,58 @@
+-module(serde_arrow_primitive_array_SUITE).
+
+-compile(export_all).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+-include("serde_arrow_primitive_array.hrl").
+
+all() ->
+    [
+        valid_len_on_new,
+        valid_null_count_on_new,
+        valid_validity_bitmap_on_new,
+        valid_len_on_access,
+        valid_null_count_on_access,
+        valid_validity_bitmap_on_access
+    ].
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Primitive Array Creation Tests %%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+valid_len_on_new(_Config) ->
+    Array = serde_arrow_primitive_array:new([1, 2, 3], {s, 8}),
+    ?assertEqual(Array#primitive_array.len, 3).
+
+valid_null_count_on_new(_Config) ->
+    Array1 = serde_arrow_primitive_array:new([1, 2, 3], {s, 8}),
+    ?assertEqual(Array1#primitive_array.null_count, 0),
+
+    Array2 = serde_arrow_primitive_array:new([1, nil, 2, 3], {s, 8}),
+    ?assertEqual(Array2#primitive_array.null_count, 1).
+
+valid_validity_bitmap_on_new(_Config) ->
+    Array1 = serde_arrow_primitive_array:new([1, 2, 3], {s, 8}),
+    ?assertEqual(Array1#primitive_array.validity_bitmap, nil),
+
+    Array2 = serde_arrow_primitive_array:new([1, nil, 2, 3], {s, 8}),
+    ?assertEqual(Array2#primitive_array.validity_bitmap, [
+        <<0:1, 0:1, 0:1, 0:1, 1:1, 1:1, 0:1, 1:1>>
+    ]).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Primitive Array Data and Metadata Access Tests %%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+valid_len_on_access(_Config) ->
+    Array = serde_arrow_primitive_array:new([1, 2, 3], {s, 8}),
+    ?assertEqual(serde_arrow_primitive_array:len(Array), 3).
+
+valid_null_count_on_access(_Config) ->
+    Array = serde_arrow_primitive_array:new([1, 2, 3], {s, 8}),
+    ?assertEqual(serde_arrow_primitive_array:null_count(Array), 0).
+
+valid_validity_bitmap_on_access(_Config) ->
+    Array = serde_arrow_primitive_array:new([1, 2, 3], {s, 8}),
+    ?assertEqual(serde_arrow_primitive_array:validity_bitmap(Array), nil).
```

Comment:
Should I just test `new/1` and use the accessor functions instead of using the record syntax, or is this fine?

Reply:
I'm assuming that this okay for now... 

I'm asking because some reviewers prefer not testing accessor functions like this because unless you have don't something monumentally stupid, it's bound to work. The reason I'm testing it anyway is because I think I'm capable of doing something monumentally stupid in the future.

Reply:
This is good, no reason not to test them.